### PR TITLE
BUG: select_as_multiple doesn't respect start/stop kwargs GH16209

### LINKED ISSUE #16209: BUG: HDFStore.select_as_multiple doesn't respect start/stop kwargs
```python
import pandas as pd

df = pd.DataFrame({"foo": [1, 2], "bar": [1, 2]})

with pd.HDFStore("foo.h5", 'w') as store:
    store.append_to_multiple({'selector': ['foo'], 'data': None}, df, selector='selector')
    single_row = store.select_as_multiple(['selector', 'data'], selector='selector', start=0, stop=1)
    assert len(single_row) == 1, "requested 1 row, got back {}".format(len(single_row))


```
Currently select_as_multiple returns the entire table. I would expect, if start=0 and stop=1, for just one row to be returned

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.32-642.15.1.el6.centos.plus.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.19.2
nose: None
pip: 9.0.1
setuptools: 32.3.1.post20170108
Cython: 0.25.2
numpy: 1.11.3
scipy: 0.18.1
statsmodels: 0.8.0
xarray: 0.8.2
IPython: 5.1.0
sphinx: 1.5.3
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.2.0
tables: 3.4.0
numexpr: 2.6.1
matplotlib: None
openpyxl: None
xlrd: 1.0.0
xlwt: None
xlsxwriter: 0.9.6
lxml: None
bs4: 4.5.3
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.1.5
pymysql: 0.7.9.None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by JosephWagner**
- [x] closes #16209
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/test_pytables.py [hunk 1]
```diff
@@ -4186,6 +4186,20 @@ def test_start_stop_table(self):
             expected = df.loc[30:40, ['A']]
             tm.assert_frame_equal(result, expected)
 
+    def test_start_stop_multiple(self):
+
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: can you add the issue number as a comment

File: pandas/tests/io/test_pytables.py [hunk 2]
```diff
@@ -4186,6 +4186,20 @@ def test_start_stop_table(self):
...
+        with ensure_clean_store(self.path) as store:
+
+            df = DataFrame({"foo": [1, 2], "bar": [1, 2]})
+
+            store.append_to_multiple({'selector': ['foo'], 'data': None}, df,
+                                     selector='selector')
+            result = store.select_as_multiple(['selector', 'data'],
+                                              selector='selector', start=0,
+                                              stop=1)
+            expected = df[['foo', 'bar']].iloc[[0]]
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: ``df.loc[[0], ['foo', 'bar']]`` is more idiomatic

File: doc/source/whatsnew/v0.21.0.txt [hunk 3]
```diff
@@ -82,6 +82,7 @@ Indexing
 I/O
 ^^^
 
+- Bug in ``HDFStore.select_as_multiple()`` where start/stop arguments were not respected (:issue:`16209`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: you can put this in 0.20.2

**Comment by jreback** [comment]
minor comments, lgtm. ping on green.

**Comment by TomAugspurger** [comment]
@JosephWagner if you can update according to Jeff's comments we can get this in for 0.20.2 (end of the week). OK to leave for 0.21.0 though.

File: doc/source/whatsnew/v0.21.0.txt (hunk 3 above)
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: can you move this

File: pandas/tests/io/test_pytables.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: can you add the issue number

**Comment by jreback** [comment]
this lgtm. Though if you can update to comments can merge for 0.20.2.

**Comment by JosephWagner** [comment]
Sorry for the late response. I've updated the PR as specified. Let me know if more changes are needed!

**Comment by jreback** [comment]
thanks!

**Pull request merged**